File: BiOS/MainViewController.m
//
//  MainViewController.m
//  BiOS
//
//  Created by BiaoWu on 16/8/9.
//  Copyright © 2016年 BiaoWu. All rights reserved.
//

#import "MainModel.h"
#import "MainViewController.h"

@implementation MainViewController

MainModel *model;

- (void)viewDidLoad {
  model = [[MainModel alloc] init];
}

- (NSInteger)tableView:(UITableView *)tableView
 numberOfRowsInSection:(NSInteger)section {
  return model.size;
}

- (UITableViewCell *)tableView:(UITableView *)tableView
         cellForRowAtIndexPath:(NSIndexPath *)indexPath {
  UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell"];

  cell.textLabel.text = [model getItem:indexPath.row];

  return cell;
}

- (void)tableView:(UITableView *)tableView
    didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
  [self showViewController:[model providerVc:indexPath.row] sender:nil];
}

@end

File: BiOS/MainViewController.swift
//
//  MainViewController.swift
//  BiOS
//
//  Created by BiaoWu on 16/8/9.
//  Copyright © 2016年 BiaoWu. All rights reserved.
//

import UIKit

class MainViewController: UITableViewController {
    
    var model = MainModel()
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return model.size()
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: "cell")!
        
        cell.textLabel?.text = model.getItem((indexPath as NSIndexPath).row).title
        
        return cell
    }
    
    override func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        show(model.providerVc((indexPath as NSIndexPath).row), sender: nil)
    }
}

